src/main/java/com/blackduck/integration/sca/upload/client/uploaders/BinaryUploader.java

```diff
@@ -136,7 +136,12 @@ protected ThrowingFunction<Response, BinaryUploadStatus, IntegrationException> c
 
             Map<String, String> responseHeaders = response.getHeaders();
             String location = Optional.ofNullable(responseHeaders.get(HttpHeaders.LOCATION)).orElseThrow(() -> new IntegrationException("Could not find Location header."));
-            String eTag = Optional.ofNullable(responseHeaders.get(HttpHeaders.ETAG)).orElseThrow(() -> new IntegrationException("Could not find Etag header."));
+            String eTag = responseHeaders.entrySet().stream()
+                .filter(entry -> entry.getKey().equalsIgnoreCase("ETag")) // Case-insensitive match
```

Comment:
Only slight suggestion, but I think we should still the HttpHeaders.ETAG enum here in place of a string literal. Likewise, the comment may not be needed as I think it's pretty clear what is being checked
```suggestion
                .filter(entry -> entry.getKey().equalsIgnoreCase(HttpHeaders.ETAG))
```

Reply:
Yes, using an enum seems like a better choice. I will update it.